pkg/provider/deployments.go

```diff
@@ -65,6 +66,30 @@ func (d *Deployment) ToString() string {
 	)
 }
 
+func (d *Deployment) AffinityRequired() bool {
+	if val, ok := d.Labels[AffinityRequiredKey]; ok {
+		result, err := strconv.ParseBool(val)
+		if err != nil {
+			return false
+		}
+		return result
+	}
+	return false
+}
+
+func (d *Deployment) IsAffinityCompliant() error {
+	var errMsg error
+	//nolint:gocritic
```

Comment:
Gocritic can be avoided easily changing this if/else if/else if for a swich, or just three ifs with a fast return:
```
func (d *Deployment) IsAffinityCompliant() error {
	if d.Spec.Template.Spec.Affinity == nil {
		return fmt.Errorf("%s has been found with an AffinityRequired flag but is missing corresponding affinity rules", d.String())
	}

	if d.Spec.Template.Spec.Affinity.PodAntiAffinity != nil {
		return fmt.Errorf("%s has been found with an AffinityRequired flag but has anti-affinity rules", d.String())
	}

	if d.Spec.Template.Spec.Affinity.NodeAffinity == nil {
		return fmt.Errorf("%s has been found with an AffinityRequired flag but is missing corresponding pod/node affinity rules", d.String())
	}

	return nil
}
```
Nit 1: The existence of this method in the Deployment struct implicitly says that all deployments should have an Affinity field that satisfies this checks. But that's only the case for deployments with the "AffinityRequired" label. Why not checking that label here and do the appropriate affinity checks based on it? Same for stateulfSets and pods.
 
Nit 2: With  IsX() functions I usually expect a `bool` as a return param. In this case, I would have done this:
`IsAffinityCompliant() (isCompliant bool, nonCompliantReason string)`

Reply:
Done.